install.sh

```diff
@@ -1,90 +1,349 @@
-echo "$(tput bold)$(tput setaf 1)If you need Card Vault, please create master key by following below steps, leave it empty if you don't need it$(tput sgr0)"
-echo "$(tput bold)$(tput setaf 3)To generate the master key, you can use the utility bundled within \n(https://github.com/juspay/hyperswitch-card-vault)$(tput sgr0)"
-echo "$(tput bold)$(tput setaf 3)If you have cargo installed you can run \n(cargo install --git https://github.com/juspay/hyperswitch-card-vault --bin utils --root . && ./bin/utils master-key && rm ./bin/utils && rmdir ./bin)$(tput sgr0)"
-
-echo "Please input the encrypted master key (optional): "
-read -s MASTER_KEY
+echo "Dependency installation completed."
+
+echo
+echo "${blue}##########################################${reset}"
+echo "${blue}    Checking neccessary permissions${reset}"
+echo "${blue}##########################################${reset}\n"
+echo
+
+check_root_user() {
+  AWS_ARN=$(aws sts get-caller-identity --output json | jq -r .Arn )
+  if [[ $AWS_ARN == *":root"* ]]; then
+    echo "ROOT user is not recommended. Please create a new user with AdministratorAccess and use their Access Token."
+    exit 1
+  fi
+}
+
+REQUIRED_POLICIES=("AdministratorAccess") # Add other necessary policies to this array
+# Check if the current user is a root user
+echo "Checking if the current AWS user is root..."
+(check_root_user) & show_loader "Verifying root user status"
+
+check_iam_policies() {
+  USER_POLICIES=$(aws iam list-attached-role-policies --role-name "$AWS_ROLE" --output json | jq -r '.AttachedPolicies[].PolicyName')
+  for policy in "${REQUIRED_POLICIES[@]}"; do
+    if ! echo "$USER_POLICIES" | grep -q "$policy"; then
+      echo "Required policy $policy is not attached to your user. Please attach this policy."
+      exit 1
+    fi
+  done
+  echo "All necessary permissions are in place."
+}
+
+# Check for specific IAM policies
+echo "Checking for necessary IAM policies..."
+(check_iam_policies) & show_loader "Verifying IAM policies"
+
+echo
+echo "${blue}##########################################${reset}"
+echo "${blue} Configure Credentials of the Application ${reset}"
+echo "${blue}##########################################${reset}"
+echo
+
+# Function to validate the password
+validate_password() {
+    if [[ ! $1 =~ ^([A-Z]|[a-z])([A-Z]|[a-z]|[0-9]){7,}$ ]]; then
+        display_error "Error: Input does not match the pattern [A-Z][a-z][0-9] and should have at least 8 characters and start with alphabet."
+        return 1
+    fi
+    return 0
+}
+
+# Prompt for DB Password
+while true; do
+    echo "Please enter the password for your RDS instance (Minimum 8 characters; includes [A-Z], [a-z], [0-9]): "
+    read -r -s DB_PASS
+    if validate_password "$DB_PASS"; then
+        break
+    fi
+done
+
+validate_api_key() {
+    if [[ ! $1 =~ ^([A-Z]|[a-z]|[0-9]|_){8,}$ ]]; then
+        display_error "Error: API Key must be at least 8 characters long and can include letters, numbers, and underscores."
```

Comment:
```suggestion
    if [[ ! $1 =~ ^([A-Za-z0-9_){8,}$ ]]; then
        display_error "Error: API Key must be at least 8 characters long and can include letters, numbers, and underscores."
```

Also, the (admin) API key can consist of any set of characters that are valid in a HTTP header.